elements/drag-drop-elements/pl-drag-drop-element.py

```diff
@@ -0,0 +1,389 @@
+
+    for html_tags in element:
+        if html_tags.tag == 'pl-answer':
+            isCorrect = pl.get_string_attrib(html_tags, 'correct', 'true')  # default to true, for backward compatibility
+            answerIndent = pl.get_string_attrib(html_tags, 'indent', '-1')  # get answer indent, and default to -1 (indent level ignored)
+            if isCorrect.lower() == 'true':
+                # add option to the correct answer array, along with the correct required indent
+                correct_answers.append(str.strip(html_tags.text))
+                correct_answers_indent.append(answerIndent)
+
+    data['params'][answerName] = mcq_options
+    data['correct_answers'][answerName] = {'correct_answers': correct_answers,
+                                           'correct_answers_indent': correct_answers_indent}
+
+
+def parse(element_html, data):
+    element = lxml.html.fragment_fromstring(element_html)
+    answerName = pl.get_string_attrib(element, 'answers-name')
+
+    temp = answerName
+    temp += '-input'  # this is how the backend is written
+
+    student_answer_temp = ''
+    if temp in data['raw_submitted_answers']:
+        student_answer_temp = data['raw_submitted_answers'][temp]
+
+    if student_answer_temp is None:
+        data['format_errors'][answerName] = 'NULL was submitted as an answer!'
+        return
+    elif student_answer_temp == '':
+        data['format_errors'][answerName] = 'No answer was submitted.'
+        return
+
+    student_answer_temp = list(student_answer_temp.rsplit(',,,'))
+
+    student_answer = []
+    student_answer_indent = []
+    permutationMode = pl.get_string_attrib(element, 'permutation-mode', 'html-order')
+
+    student_answer_ranking = ['Question permutationMode is not "ranking"']
+    for answer in student_answer_temp:
+        # student answers are formatted as: {answerString}:::{indent}, we split the answer
+        answer = answer.rsplit(':::')
+        if len(answer) == 1:
+            # because we already caught empty string submission above
+            # failing to split the answer implies an error
+            data['format_errors'][answerName] = 'Failed to parse submission: formatting is invalid! This should not happen, contact instructor for help.'
+            return
+
+        if (not str.isdigit(answer[1]) and answer[1] != '-1') or int(answer[1]) not in list(range(-1, 5)):  # indent is a number in [-1, 4]
+            data['format_errors'][answerName] = f'Indent level {answer[1]} is invalid! Indention level must be a number between 0 or 4 inclusive.'
+            return
+
+        student_answer.append(answer[0])
+        student_answer_indent.append(answer[1])
+    del student_answer_temp
+
+    if permutationMode.lower() == 'ranking':
+        student_answer_ranking = []
+        pl_drag_drop_element = lxml.html.fragment_fromstring(element_html)
+        for answer in student_answer:
+            e = pl_drag_drop_element.xpath(f'.//pl-answer[text()="{answer}"]')
```

Comment:
As I said in a comment, this seems to not be working when there is whitespace on the sides of the answer in the question. 

Also, using `xpath` here seems like total over kill. Why not just iterate over the `element_html` to get the string from each of the possible answers, then compare against those directly?

Reply:
yeah, I remember I did try another approach, but I found that using `xpath` was a quick and dirty fix, and then I left it in

this probably happened a good number of weeks ago, I'll be looking at this